### PR TITLE
Create new actions for all workload resources

### PR DESCRIPTION
**Pull request by knnguy**
Created new actions for:

- Pods
- Deployments
- Services
- Namespaces

Added delete actions for:
- StatefulSets
- ReplicaSets
- DaemonSets
- Jobs
- Cronjobs

### CODE REVIEW

File: src/app/plugins/kubernetes/components/cronjobs/actions.ts [hunk 1]
```diff
@@ -83,6 +84,23 @@ export const createJob = jobActions.add(
   ),
 )
 
+export const deleteJob = jobActions.add(
+  new DeleteAction<
+    DataKeys.Jobs,
+    { clusterId: string; namespace: string; name: string; id: string },
+    any
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/cronjobs/actions.ts:91]: no need for this `any`

File: src/app/plugins/kubernetes/components/deployments/new-actions.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+)
+
+listDeployments.addDependency(listPods)
+listDeployments.addDependency(listServices)
+
+export const createService = deploymentActions.add(
+  new CreateAction<
+    DataKeys.Deployments,
+    { clusterId: string; namespace: string; yaml: string },
+    any
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/deployments/new-actions.ts:48]: no need for this any, just add the correct typings in ` this.client.basicPost<GetClusterDeploymentsItem>` and it should work

File: src/app/plugins/kubernetes/components/namespaces/new-actions.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      return someAsync(pluck<any, string>('uuid', clusters).map(qbert.getClusterNamespaces)).then(
+        flatten,
+      )
+    }
+    return qbert.getClusterNamespaces(clusterId)
+  }),
+)
+
+export const createNamespace = namespaceActions.add(
+  new CreateAction<DataKeys.Namespaces, { clusterId: string; name: string }, any>(
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/namespaces/new-actions.ts:40]: Same thing here

File: src/app/plugins/kubernetes/components/pods/new-actions.ts [hunk 4]
```diff
@@ -6,19 +6,25 @@ import { flatten, pluck } from 'ramda'
...
 
 const { qbert } = ApiClient.getInstance()
 
-const podsActionSet = new ActionsSet<DataKeys.Pods>({
-  uniqueIdentifier: 'metadata.uid',
+export const podActions = new ActionsSet<DataKeys.Pods>({
+  uniqueIdentifier: 'id',
   indexBy: 'clusterId',
-  entityName: 'Pods',
+  entityName: entityNamesByKey.Pods,
```
Comment (xantorres) [inline src/app/plugins/kubernetes/components/pods/new-actions.ts:23]: We should just use the `DataKeys`, I think `entityNamesByKey` is just for legacy purposes (to be used in js files)

File: src/app/plugins/kubernetes/components/deployments/new-actions.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    {
+      clusterId: string
+      namespace: string
+      name: string
+      body: unknown
+    }
+  >(async (params) => {
+    const { clusterId, namespace, name, body } = params
+    Bugsnag.leaveBreadcrumb('Attempting to update cluster deployment', params)
+    const updatedDeployment = await qbert.updateDeployment(clusterId, namespace, name, body)
```
[the remark this answers is not in the document]
Comment (jruddell) [reply src/app/plugins/kubernetes/components/deployments/new-actions.ts:79]: the deployment is a k8s resource so I believe they return back the model

**Review by jruddell — APPROVED** [review]

**Pull request merged**